### PR TITLE
fix a11y nested span bug in psammead-bulletin

### LINKED ISSUE #9561: TalkBack incorrectly reading links in nested spans
**Describe the bug**
When using TalkBack on Android, links with role text and nested spans aren't read as one item as intended to be. Content is also repeated in additional blocks/gestures.

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to https://www.bbc.com/portuguese/podcasts/p07r3r3t
2. Navigate to the 'O podcast está disponível em' header on TalkBack
3. Swipe to the list on TalkBack
4. TalkBack should read 'Spotify, Que Historia, externo, link'
5. Swipe through the links and TalkBack should read an additional 'Spotify', and the text highlighted
6. Swipe again and TalkBack will read out 'Que Historia' as well as highlight a separate 'block' on the screen.

**Expected behaviour**
TakBack should only read out 'Spotify, Que Historia, externo, link' before moving on to the next link.

**Screenshots**
image

**Smartphone (please complete the following information):**

- Device: Samsung S10
- OS: Android 11
- Browser: Chrome
- Version: 94.0.4606.85

**Testing notes**

- [x] This bug fix is expected to need manual testing.

**Checklist**
- [x] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### PR DESCRIPTION
**Pull request by DarioR01**
Resolves [#9561](https://github.com/bbc/simorgh/issues/9561)

Fix a11y bug causing talkback needing multiple gesture to read every span inside a parent with role=text (This should only require 1 gesture) 

**Code changes:**

- added id to span with role=text
- added span with id to children
- added aria-labelledby to anchor element

---

- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-bulletin/src/__snapshots__/index.test.jsx.snap [hunk 1]
```diff
@@ -279,10 +279,12 @@ exports[`Bulletin should render audio correctly 1`] = `
         dir="ltr"
       >
         <a
+          aria-labelledby="bulletin-https://bbc.co.uk"
```
Comment (greenc05) [inline packages/components/psammead-bulletin/src/__snapshots__/index.test.jsx.snap]: I guess this is just an example? Having a url in an id looks rather strange...

**Comment by greenc05** [comment]
Looks good thanks @DarioR01

File: packages/components/psammead-bulletin/src/__snapshots__/index.test.jsx.snap (hunk 1 above)
Comment (DarioR01) [reply packages/components/psammead-bulletin/src/__snapshots__/index.test.jsx.snap]: Yes, I hope there is a better id in Simorgh

**Review by amoore108 — APPROVED** [review]

**Review by mfonofm — APPROVED** [review]

File: packages/components/psammead-bulletin/src/index.jsx [hunk 2]
```diff
@@ -203,20 +204,21 @@ const Bulletin = ({
...
                 ariaHidden
                 offScreenText={offScreenText}
+                id={`bulletin-${ariaId}`}
               >
                 {headlineText}
               </LiveLabel>
             ) : (
               // eslint-disable-next-line jsx-a11y/aria-role
-              <span role="text">
+              <span role="text" id={`bulletin-${ctaLink}`}>
```
Comment (ryanmccombe) [inline packages/components/psammead-bulletin/src/index.jsx]: URLs contain characters that can't be in IDs sadly - see this issue for more info: https://github.com/bbc/simorgh/issues/9696

One option would be to use regex to clean up the URL - something like `const id = ctaLink.replace(/\W/g, '')` should remove everything except a-z, A-Z and 0-9

**Review by ryanmccombe — CHANGES_REQUESTED** [review]

File: packages/components/psammead-bulletin/src/index.jsx (hunk 2 above)
Comment (amoore108) [reply packages/components/psammead-bulletin/src/index.jsx]: @DarioR01 I've implemented this in another area https://github.com/bbc/psammead/pull/4602/files#diff-cf9b2acec38beebd665b213ba1986338507662983a46b941657c8bbd6bfaac2dR57-R70, so hopefully it should be straightforward enough here

Comment (DarioR01) [reply packages/components/psammead-bulletin/src/index.jsx]: Yes, sorry for the late reply to everyone. It seems like I missed this from last week. I will sanitise the link after the a11y swarm 👍. Thank you @amoore108 for the implementation example 😄

**Review by ryanmccombe — APPROVED** [review]

**Pull request merged**